### PR TITLE
fix: `ColumnReader.loadVector` should initiate `CometDictionary` after re-import arrays

### LINKED ISSUE #474: ColumnReader.loadVector should initiate CometDictionary after re-import arrays
### Describe the bug

During debugging some test failures in #437, I found that `CometScanExec` returns empty dictionary values after first batch in failed queries. It is because in `Data.importVector` API call, it will release existing dictionary vector in the given `dictionaryProvider` and re-import the dictionary vector.

Currently we only initiate `CometDictionary` at the first batch and reuse it for later batches. Because of above issue, we should already initiate `CometDictionary`. It is simply re-importing existing dictionary vector allocated by native side, so it doesn't incur re-allocation of memory.

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #474.

### CODE REVIEW

**Comment by viirya** [comment]
cc @sunchao

File: common/src/main/java/org/apache/comet/parquet/ColumnReader.java [hunk 1]
```diff
@@ -230,15 +230,15 @@ public CometDecodedVector loadVector() {
...
-        dictionary = new CometDictionary(dictionaryVector);
       }
 
+      // We should already re-initiate `CometDictionary` here because `Data.importVector` API will
+      // release the previous dictionary vector and create a new one.
+      Dictionary arrowDictionary = dictionaryProvider.lookup(dictionaryEncoding.getId());
+      CometPlainVector dictionaryVector =
+          new CometPlainVector(arrowDictionary.getVector(), useDecimal128, isUuid);
+      dictionary = new CometDictionary(dictionaryVector);
+
```
Comment (viirya) [inline common/src/main/java/org/apache/comet/parquet/ColumnReader.java:241]: The test failure is in TPCDSQuerySuite in #437. I don't have a simple reproducible test case for it. But I think the change is straightforward and safe.

**Review by andygrove — APPROVED** [review]

**Review by sunchao — APPROVED** [review]

**Comment by viirya** [comment]
Merged. Thanks @andygrove @sunchao

**Comment by andygrove** [comment]
@viirya Just fyi, this commit caused a large regression for me in local runs of TPC-H. I will share full details tomorrow.

## before 

Screenshot from 2024-05-27 13-23-16

## after

Screenshot from 2024-05-27 13-23-31

**Comment by viirya** [comment]
I got the reason why it has regression. I will propose a fix.

**Pull request merged**